### PR TITLE
Fix Cloud Slack Dev E2E tests after Botkube release

### PR DESCRIPTION
**Pull request by pkosiec**
## Description

Changes proposed in this pull request:

- Fix Cloud Slack Dev E2E tests after Botkube release
- Take into account the "upgrade checker" message

## Testing

No need, the pipeline tests it: https://github.com/kubeshop/botkube/actions/runs/6457080833

### CODE REVIEW

File: .github/workflows/branch-build.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - fix-slack-e2e # TODO: Remove before merge
```
Comment (pkosiec) [inline .github/workflows/branch-build.yml:7]:
```suggestion
```

File: .github/workflows/branch-build.yml [hunk 2]
```diff
@@ -27,9 +28,10 @@ jobs:
           fetch-depth: 0
       - name: Extract version
         id: extract-version
+        # TODO: Revert
         run: |
           IMAGE_VERSION=$(git rev-parse --short HEAD)
-          echo "versions={\"image-version\":[\"v9.99.9-dev\",\"0.0.0-${IMAGE_VERSION}\"]}" >> $GITHUB_OUTPUT
+          echo "versions={\"image-version\":[\"0.0.0-${IMAGE_VERSION}\"]}" >> $GITHUB_OUTPUT
```
Comment (pkosiec) [inline .github/workflows/branch-build.yml:34]:
```suggestion
          echo "versions={\"image-version\":[\"v9.99.9-dev\",\"0.0.0-${IMAGE_VERSION}\"]}" >> $GITHUB_OUTPUT
```

File: .github/workflows/branch-build.yml [hunk 3]
```diff
@@ -27,9 +28,10 @@ jobs:
           fetch-depth: 0
       - name: Extract version
         id: extract-version
+        # TODO: Revert
```
Comment (pkosiec) [inline .github/workflows/branch-build.yml:31]:
```suggestion
```

**Review by pkosiec — COMMENTED** [review]
To revert before merge

**Review by mszostok — APPROVED** [review]

**Pull request merged**